src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -1400,138 +1400,150 @@ void TaskSketcherElements::updateVisibility()
-                            // by element number.
-                            for (int i = 0; i < countItems; i++) {
-                                ElementItem* item =
-                                    static_cast<ElementItem*>(ui->listWidgetElements->item(i));
-                                if (item->ElementNbr == ElementId) {
-                                    item->isLineSelected = select;
-                                    modified_item = item;
-                                    SketcherGui::scrollTo(ui->listWidgetElements, i, select);
-                                    break;
-                                }
-                            }
-                        }
-                    }
+    // update the listwidget
+    auto updateListWidget = [this](auto& modified_item) {
+        QSignalBlocker sigblk(this->ui->listWidgetElements);
+        if (modified_item == nullptr) {
+            return;
+        }
+        bool is_selected = modified_item->isSelected();
+        const bool should_be_selected = modified_item->isLineSelected
+            || modified_item->isStartingPointSelected || modified_item->isEndPointSelected
+            || modified_item->isMidPointSelected;
+
+        // If an element is already selected and a new subelement gets selected
+        // (eg., if you select the arc of a circle then select the center as
+        // well), the new subelement won't get highlighted in the list until you
+        // mouseover the list.  To avoid this, we deselect first to trigger a
+        // redraw.
+        if (should_be_selected && is_selected) {
+            modified_item->setSelected(false);
+            is_selected = false;
+        }
+
+        if (should_be_selected != is_selected) {
+            modified_item->setSelected(should_be_selected);
+        }
+    };
+
+    switch (msg.Type) {
+        case Gui::SelectionChanges::ClrSelection: {
+            clearWidget();
+            return;
+        }
+        case Gui::SelectionChanges::AddSelection:
+        case Gui::SelectionChanges::RmvSelection: {
+            bool select = (msg.Type == Gui::SelectionChanges::AddSelection);
+            // is it this object??
+            if (strcmp(msg.pDocName, sketchView->getSketchObject()->getDocument()->getName()) != 0
+                || strcmp(msg.pObjectName, sketchView->getSketchObject()->getNameInDocument())
+                    != 0) {
+                return;
+            }
+            if (!msg.pSubName) {
+                return;
+            }
+            ElementItem* modified_item = nullptr;
+            QString expr = QString::fromLatin1(msg.pSubName);
+            std::string shapetype(msg.pSubName);
+            // if-else edge vertex
+            if (boost::starts_with(shapetype, "Edge")) {
+                QRegularExpression rx(QStringLiteral("^Edge(\\d+)$"));
+                QRegularExpressionMatch match;
+                boost::ignore_unused(expr.indexOf(rx, 0, &match));

```

Comment:
I don't think boost::ignore_unused needed here. According to qt's docs for qstring, indexOf doesn't have [[nodiscard]].
But maybe I'm missing something?

Reply:
This was added here: 3e58513c24e533326906be5c87c82aafe582c936. I'm not versed with this method so no comments for now.

Reply:
I see.. we can look over those at a later time for the entire code base instead.